types/servercontext.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+	"os"
+)
+
+// server context
+type ServerContext struct {
+	Config *cfg.Config
+	Logger log.Logger
+}
+
+func NewDefaultServerContext() *ServerContext {
+	return NewServerContext(
+		cfg.DefaultConfig(),
+		log.NewTMLogger(log.NewSyncWriter(os.Stdout)),
+	)
+}
+
+func NewServerContext(config *cfg.Config, logger log.Logger) *ServerContext {
+	return &ServerContext{config, logger}
+}
```

Comment:
This should not be getting moved from the server package TO the types package, the whole plan is to move structs out of the types package and close to their main usage... aka `server/`. Also this unnecessarily increases the length of the type (could just be `server.Context`)

Reply:
If anything we could add an alias in the `types` package which references the `server` type